sdk/translation/ai-translation-text/src/customClient.ts

```diff
@@ -0,0 +1,100 @@
+// Licensed under the MIT license.
+
+import { getClient, ClientOptions } from "@azure-rest/core-client";
+import * as coreRestPipeline from "@azure/core-rest-pipeline";
+import {
+  PipelineRequest,
+  PipelineResponse,
+  SendRequest
+} from "@azure/core-rest-pipeline";
+import { TextTranslationClient } from "./generated/clientDefinitions";
+import {
+  TranslatorCredential,
+  TranslatorAuthenticationPolicy,
+  TranslatorAzureKeyAuthenticationPolicy
+} from "./authentication";
+import {
+  AzureKeyCredential,
+  TokenCredential
+} from "@azure/core-auth";
+
+const DEFAULT_SCOPE = "https://cognitiveservices.azure.com/.default";
+const DEFAULT_ENPOINT = "https://api.cognitive.microsofttranslator.com";
+const PLATFORM_HOST = "cognitiveservices";
+const PLATFORM_PATH = "/translator/text/v3.0";
+
+/**
+ * Initialize a new instance of `TextTranslationClient`
+ * @param endpoint type: string, Supported Text Translation endpoints (protocol and hostname, for example:
+ *     https://api.cognitive.microsofttranslator.com).
+ * @param options type: ClientOptions, the parameter for all optional parameters
+ */
+export default function createClient(
+  endpoint: undefined | string,
+  credential: undefined | TranslatorCredential | AzureKeyCredential | TokenCredential = undefined,
+  options: ClientOptions = {}
+): TextTranslationClient {
+  let serviceEndpoint: string;
+  if (!endpoint) {
+    serviceEndpoint = DEFAULT_ENPOINT;
+  } else if (endpoint.toLowerCase().indexOf(PLATFORM_HOST) != -1) {
+    serviceEndpoint = `${endpoint}${PLATFORM_PATH}`;
+  } else {
+    serviceEndpoint = endpoint;
+  }
+
+  const baseUrl = options.baseUrl ?? `${serviceEndpoint}`;
+
+  const userAgentInfo = `azsdk-js-ai-translation-text-rest/1.0.0-beta.1`;
+  const userAgentPrefix =
+    options.userAgentOptions && options.userAgentOptions.userAgentPrefix
+      ? `${options.userAgentOptions.userAgentPrefix} ${userAgentInfo}`
+      : `${userAgentInfo}`;
+  options = {
+    ...options,
+    userAgentOptions: {
+      userAgentPrefix,
+    },
+  };
+
+  const client = getClient(baseUrl, options) as TextTranslationClient;
+  client.pipeline.addPolicy(apiVersionPolicy);
+
+  if (credential instanceof TranslatorCredential) {
+    const mtAuthneticationPolicy = new TranslatorAuthenticationPolicy(credential as TranslatorCredential);
+    client.pipeline.addPolicy(mtAuthneticationPolicy);
+  } else if (credential instanceof AzureKeyCredential) {
+    const mtKeyAuthenticationPolicy = new TranslatorAzureKeyAuthenticationPolicy(credential as AzureKeyCredential);
+    client.pipeline.addPolicy(mtKeyAuthenticationPolicy);
+  } else if (credential) {
+    client.pipeline.addPolicy(
+      coreRestPipeline.bearerTokenAuthenticationPolicy({
+        credential: credential as TokenCredential,
+        scopes: DEFAULT_SCOPE
+      })
+    );
+  }
+
+  return client;
+}
+
+/** Policy that sets the api-version (or equivalent) to reflect the library version. */
+const apiVersionPolicy = {
```

Comment:
I wonder why is this needed? the swagger/typespec document should model the api version and the generated client should pick up the default one. This code will make this library not work with any other API version in the future.

Reply:
I believe it was added because of this: https://teams.microsoft.com/l/message/19:4844af8fce9742a98d06e1debf60bab2@thread.skype/1676589603041?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1676589603041&teamName=Azure%20SDK&channelName=DPG%20Support&createdTime=1676589603041

I can see that the open issue is not resolved. We can revisit this for GA.

Reply:
@MikeyMCZ I think the issue for translation should have been resolved, you can try our latest emitter https://www.npmjs.com/package/@azure-tools/typespec-ts?activeTab=versions 
The reason this issue is still open is because there are some other cases we need to resolve. @MaryGao Please correct me if I am wrong.

Reply:
ACK, we will use the latest emitter for the next PR and we will remove the apiVersionPolicy. Thank you!